### PR TITLE
docs - add content for new 6x gp_parallel_retrieve_cursor module (beta)

### PR DESCRIPTION
**Pull request by lisakowen**
docs for https://github.com/greenplum-db/gpdb/pull/12645, which introduces a new gp_parallel_retrieve_cursor module. in this PR:
- doc the new module.
- some misc nav updates.
 
(there will be a separate PR for master, as the parallel cursor functionality is built-in there, not provided in a module.)

doc review site link (behind vpn):  https://docs-lisa-prc.sc2-04-pcf1-apps.oc.vmware.com/7-0/ref_guide/modules/gp_parallel_retrieve_cursor.html. **NOTE:** this review site includes the doc updates from the open PR https://github.com/greenplum-db/gpdb/pull/13134 so that the links to the new views/commands will work.

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+                  to the <codeph>RETRIEVE</codeph> command.</entry>
+              </row>
+              </tbody>
+            </tgroup>
+          </table>
+          <p>Refer to the <xref href="../system_catalogs/gp_endpoints.html#topic1" format="html"
+               scope="peer">gp_endpoints</xref> view reference page for more information
+            about the endpoint attributes returned by these commands.</p>
+          <p>You can similarly invoke the <codeph>gp_get_session_endpoints()</codeph>
+            function or examine the <codeph>gp_session_endpoints()</codeph> view
```
Comment (adam8157) [inline gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml:247]: view has no `()`

**Review by adam8157 — COMMENTED** [review]
Others look good.

File: gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+            <codeph>gp_get_segment_endpoints()</codeph> function or examining the
+            <codeph>gp_segment_endpoints</codeph> view:</p>
+          <codeblock>SELECT * FROM gp_get_segment_endpoints();
+SELECT * FROM gp_segment_endpoints;</codeblock>
+          <p>These commands list the endpoints
+            created in the QE for all active parallel retrieve cursors declared by the
+            current session user. When the Greenplum Database superuser invokes the
+            command, it returns a list of all endpoints on the QE created for all parallel
+            retrieve cursors declared by all users.</p>
+          <note>You must initiate a utility-mode connection to the endpoint before you run
```
Comment (soumyadeep2007) [inline gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml:373]: There is another way to see this info from the master node (QD) as well:

```
postgres=# SELECT * from gp_dist_random('gp_segment_endpoints');
            auth_token            | databaseid | senderpid | receiverpid |   state  | gp_segment_id | sessionid | username |    endpointname    | cursorname 
----------------------------------+------------+-----------+-------------+----------+---------------+-----------+----------+--------------------+------------
 2e2d095a0b25051ee81f1f6c1c5fa5cc |      12812 |   1567301 |     1574614 | ATTACHED |             0 |         8 | pivotal  | c10000000800000003 | c1
 d2e4eee3669104a74c1facbacf5e2993 |      12812 |   1567302 |          -1 | READY    |             1 |         8 | pivotal  | c10000000800000003 | c1
 d3825fc07e56bee5fcd2b1d0b600c85e |      12812 |   1567303 |          -1 | READY    |             2 |         8 | pivotal  | c10000000800000003 | c1
(3 rows)
```

One can also filter the above to one segment:
`SELECT * from gp_dist_random('gp_segment_endpoints') WHERE gp_segment_id = 0;`

Comment (soumyadeep2007) [reply gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml:373]: ^ The above query is incredibly valuable from a DBA perspective - it allows the DBA to figure out all of the endpoints, their sender pids and corresponding retrieve connection receiver pids running in the system, their states etc.

We would probably want to turn this query into a new view in the next release.

File: gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+                  retrieve cursor was created.</entry>
+              </row>
+              <row>
+                <entry colname="col1">senderpid</entry>
+                <entry colname="col2">The identifier of the process sending the query
+                  results.</entry>
+              </row>
+              <row>
+                <entry colname="col1">receiverpid</entry>
+                <entry colname="col2">The identifier of the process receiving the query
```
Comment (soumyadeep2007) [inline gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml:405]: Can we somehow mention that this process is tied to the retrieve connection?

**Review by soumyadeep2007 — APPROVED** [review]
Looks great!

File: gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml (hunk 2 above)
Comment (lisakowen) [reply gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml]: thanks for this input, @soumyadeep2007 .  i pushed a commit to add this info.  i also updated the staging site (but note that links to RETRIEVE, new views, and gp_retrieve_conn do not work, i'll rebuild a complete review site with that PR info later.)

Comment (soumyadeep2007) [reply gpdb-doc/dita/ref_guide/modules/gp_parallel_retrieve_cursor.xml]: Looks good! Thanks!

**Review by dyozie — APPROVED** [review]

**Pull request merged**